WalletWasabi/Blockchain/Mempool/MempoolService.cs

```diff
@@ -1,33 +1,21 @@
+using System.Collections;
 using NBitcoin;
 using System.Collections.Generic;
 using System.Diagnostics.CodeAnalysis;
 using System.Linq;
-using System.Threading;
-using System.Threading.Tasks;
 using WalletWasabi.Blockchain.Analysis.Clustering;
 using WalletWasabi.Blockchain.TransactionOutputs;
 using WalletWasabi.Blockchain.Transactions;
-using WalletWasabi.Logging;
+using WalletWasabi.Cache;
 using WalletWasabi.Models;
-using WalletWasabi.WebClients.Wasabi;
 
 namespace WalletWasabi.Blockchain.Mempool;
 
 public class MempoolService
 {
-	/// <summary>Denotes whether we are cleaning up the mempool at the moment or not.</summary>
-	private int _cleanupInProcess = 0;
-
-	private long _totalReceives = 0;
-	private long _duplicatedReceives = 0;
-
 	public event EventHandler<SmartTransaction>? TransactionReceived;
 
-	/// <remarks>Guarded by <see cref="ProcessedLock"/>.</remarks>
-	private HashSet<uint256> ProcessedTransactionHashes { get; } = new();
-
-	/// <summary>Guards <see cref="ProcessedTransactionHashes"/>.</summary>
-	private object ProcessedLock { get; } = new();
+	private readonly MemoryCache<uint256, bool> _cache = new(TimeSpan.FromMinutes(30));
```

Comment:
The name was chosen like that because I want to evolve it to replace the dotnet `MemoryCache`. Basically I wanted to start with the minimum viable implementation that solves all the [problems described here](https://github.com/dotnet/runtime/issues/48567) using what David Fowler suggests with a smaller but very similar api. Next step would be to implement it in a way that doesn't suffer the race conditions that the official MemoryCache has.